_posts/2018-06-21-vision-for-rust-and-wasm.md

```diff
@@ -0,0 +1,255 @@
+5. and publish the resulting package on npm.
+
+All without you, the Rust developer, needing to have a JavaScript toolchain up
+and running.
+
+Right now, steps 1, 2, and 5 are in place, but you still need to have `npm`
+installed locally. There are also some more things planned for `wasm-pack`, and
+our story for orchestrating builds, dependencies, and publishing coming down the
+pipe, but you'll have to wait for the dedicated follow up blog post.
+
+### Wait, There's More!
+
+<a href="/images/twiggy.png">
+  <img src="/images/twiggy.png" alt="Twiggy!" style="float:right;width:40%;min-width:100px;max-width:500px;margin:1em"/>
+</a>
+
+* [Twiggy is a code size profiler for `.wasm` binaries.][twiggy] It helps you
+  answer questions like "why did this function even end up in here -- who calls
+  it?" and "how much space would be saved if I stopped using this function,
+  removed it, and removed all the functions that become dead code after its
+  removal?"
+
+* [`wee_alloc` is a tiny allocator designed for WebAssembly that has a (pre
+  compression) code size footprint of only a single kilobyte.][wee_alloc] It is
+  geared towards code that makes a handful of initial dynamically sized
+  allocations, and then performs its heavy lifting without any further
+  allocations. This scenario requires *some* allocator to exist, but we are more
+  than happy to trade allocation performance for small code size.
+
+<div style="clear: both"/>
+
+<a href="/images/console_error_panic_hook.png">
+  <img src="/images/console_error_panic_hook.png" alt="Twiggy!" style="float:left;width:60%;min-width:100px;max-width:800px;margin:1em"/>
+</a>
+
+* [The `console_error_panic_hook` crate provides a panic hook for wasm that logs
+  panics to the developer console via the `console.error`
+  function.][console_error_panic_hook] No more opaque "RuntimeError: unreachable
+  executed" messages! Get the proper assertion failure message or index
+  out-of-bounds information you expect. It makes debugging panics a whole lot
+  easier.
+
+* [The `wasm-snip` tool lets you forcibly replace a function's body with a
+  single `unreachable` instruction.][wasm-snip] Maybe you know that some
+  function will never be called at runtime, but the compiler can't prove that at
+  compile time? Snip it! Then run wasm-gc again and all the functions it
+  transitively called (which could also never be called at runtime) will get
+  removed too. This is particularly helpful for removing Rust's panicking and
+  formatting infrastructure when you intend to ship small `.wasm` binaries with
+  `panic=abort`.
+
+<div style="clear: both"/>
+
+## Coming Soon: The Future
+
+As mentioned throughout this post, we'll be following up with more blog posts
+detailing specific goals we have for the Rust 2018 edition and how you can
+help. In the meantime, don't hesitate to [join the Rust and WebAssembly domain
+working group and help build the future of Rust and WebAssembly now!][get-involved]
```

Comment:
Maybe before release we create some easy issues for the things we mentioned and reserve them for first time contributors to the org. Then we can say get started today, here's where you can help for x thing